modules/home-environment.nix

```diff
@@ -115,6 +173,12 @@ in
       description = "The user's username.";
     };
 
+    home.mailAccounts = mkOption {
+      type = types.listOf mailAccount;
```

Comment:
Could use `types.loaOf` like in `fileType` (see `modules/lib/file-type.nix`) here. Then it would be possible to write something like

```
  home.mailAccounts = {
    "gmail_perso" = {
      userName = "login";
      realname = "Luke skywalker";
      address = "home-manager-rocks@gmail.com";
      store = "maildir/gmail";
      sendHost = "smtp.gmail.com";
      # getLogin = "";
      # getPass = "";
    };
  };
```